.github/workflows/multi_service_image_build.yml

```diff
@@ -54,32 +58,53 @@ jobs:
           do 
             if [[ -e $DIR/src && -e $DIR/Dockerfile ]]
             then
+              DIR=`echo $DIR| sed -r 's/\/$//'`
               CHANGED_WLS_DIRS+=("$DIR")
             fi  
           done
           echo "CHANGED_WLS_DIRS=${CHANGED_WLS_DIRS[@]}">> "$GITHUB_ENV"
-          for wl_service in ${CHANGED_WLS_DIRS[@]}
+          for wl_service_dir in ${CHANGED_WLS_DIRS[@]}
           do
+              wl_service=`basename ${wl_service_dir}`
+              if [[ ${wl_service_dir} =~ ^app ]]
+                then 
+                    DOCKERFILE="${wl_service_dir}/Dockerfile"
+                    BUILD_CONTEXT=""
+                    echo "APP_STRUCT=1">>"$GITHUB_ENV"
+                else
+                    DOCKERFILE="Dockerfile"
+                    BUILD_CONTEXT="${wl_service_dir}"
+              fi
               argo submit .argo/wl-service-wf.yaml \
                     --generate-name ${wl_service}-wf- \
-                    -p repo=git@github.com:<GIT_ORGANIZATION_NAME>/<WL_REPO_NAME>.git \
+                    -p repo=git@github.com:cgdevx-demo/vt-cnask.git \
                     -p wl-service-name=$wl_service \
-                    -p workload-name=<WL_REPO_NAME> \
+                    -p workload-name=vt-cnask \
                     -p tag=$TAG \
+                    -p dockerfile=${DOCKERFILE} \
+                    -p build-context=${BUILD_CONTEXT} \
+                    -p wl-service-dir=${wl_service_dir} \
+                    --serviceaccount argo-workflow \
                     --wait --log 
           done
       - name: version change in gitops repo
         env:
           CHANGED_WLS_DIRS: ${{ env.CHANGED_WLS_DIRS }}
+          APP_STRUCT: ${{ env.APP_STRUCT }}
         run: |-
           echo "CHANGED workload-services (array): ${CHANGED_WLS_DIRS[@]}"
           if [[ ${#CHANGED_WLS_DIRS[@]} -gt 0 ]]
           then
-           argo submit .argo/version-changer-wf.yaml \
-              -p gitops-repo=git@github.com:<GIT_ORGANIZATION_NAME>/<WL_GITOPS_REPO_NAME>.git \
+             echo "${CHANGED_WLS_DIRS[@]}"
+             if  [[ $APP_STRUCT -ne 1 ]]
+             then
+               argo submit .argo/version-changer-wf.yaml \
+              -p gitops-repo=git@github.com:cgdevx-demo/vt-cnask-gitops.git \
```

Comment:
Looks like an error here. Need use parameters  <GIT_ORGANIZATION_NAME> & <WL_GITOPS_REPO_NAME>
As for me, need to replace git@github.com with some new parameter to have support for GitLab in the future.

Reply:
probably <WL_GITOPS_REPOSITORY_GIT_URL> will work better here

Reply:
Fixed, thank you.